tajo-core/pom.xml

```diff
@@ -93,33 +93,29 @@
         </executions>
       </plugin>
       <plugin>
-        <groupId>org.codehaus.mojo</groupId>
-        <artifactId>exec-maven-plugin</artifactId>
-        <version>1.2</version>
+        <groupId>com.github.os72</groupId>
+        <artifactId>protoc-jar-maven-plugin</artifactId>
+        <version>3.5.0</version>
         <executions>
           <execution>
-            <id>generate-sources</id>
             <phase>generate-sources</phase>
-            <configuration>
-              <executable>protoc</executable>
-              <arguments>
-                <argument>-Isrc/main/proto/</argument>
-                <argument>--proto_path=../tajo-common/src/main/proto</argument>
-                <argument>--proto_path=../tajo-catalog/tajo-catalog-common/src/main/proto</argument>
-                <argument>--proto_path=../tajo-client/src/main/proto</argument>
-                <argument>--proto_path=../tajo-plan/src/main/proto</argument>
-                <argument>--java_out=target/generated-sources/proto</argument>
-                <argument>src/main/proto/ResourceTrackerProtocol.proto</argument>
-                <argument>src/main/proto/QueryMasterProtocol.proto</argument>
-                <argument>src/main/proto/QueryCoordinatorProtocol.proto</argument>
-                <argument>src/main/proto/TajoWorkerProtocol.proto</argument>
-                <argument>src/main/proto/InternalTypes.proto</argument>
-                <argument>src/main/proto/ResourceProtos.proto</argument>
-              </arguments>
-            </configuration>
             <goals>
-              <goal>exec</goal>
+              <goal>run</goal>
             </goals>
+            <configuration>
+              <protocVersion>2.5.0</protocVersion>
+              <includeStdTypes>true</includeStdTypes>
+              <inputDirectories>
+                <include>src/main/proto/</include>
+              </inputDirectories>
+              <includeDirectories>
+                <includeDirectory>../tajo-common/src/main/proto</includeDirectory>
+                <includeDirectory>../tajo-catalog/tajo-catalog-common/src/main/proto</includeDirectory>
+                <includeDirectory>../tajo-core/src/main/proto</includeDirectory>
```

Comment:
Has to be ```../tajo-client/src/main/proto```?

Reply:
Interestingly, tajo-client/src/main/proto was not necessary. The protobuf files of tajo-client depend on only those of tajo-core, but it wasn't vise versa.

Reply:
Ok, then we don't have include the input directory here as well.

Reply:
I removed that too.